### PR TITLE
Feature WYSIWYG editor

### PR DESCRIPTION
**Pull request by sosukesuzuki**
I am implementing [#236](https://github.com/BoostIO/Boostnote/issues/236).
using [react-prosemirror](https://github.com/tgecho/react-prosemirror).

### CODE REVIEW

**Comment by sosukesuzuki** [comment]
**Attention**
ProseMirror uses [commonmark](http://commonmark.org/) 💀

**Comment by kazup01** [comment]
refs: #1207

File: browser/main/Detail/MarkdownNoteDetail.js [hunk 1]
```diff
@@ -380,15 +407,7 @@ class MarkdownNoteDetail extends React.Component {
...
-          <MarkdownEditor
-            ref='content'
-            styleName='body-noteEditor'
-            config={config}
-            value={this.state.note.content}
-            storageKey={this.state.note.storage}
-            onChange={(e) => this.handleChange(e)}
-            ignorePreviewPointerEvents={this.props.ignorePreviewPointerEvents}
-          />
+          {this.renderEditor()}
```
Comment (sota1235) [inline browser/main/Detail/MarkdownNoteDetail.js]: 👍

File: browser/main/Detail/MarkdownNoteDetail.js [hunk 2]
```diff
@@ -262,6 +263,32 @@ class MarkdownNoteDetail extends React.Component {
     ee.emit('print')
   }
 
+  renderEditor () {
+    const { config } = this.props
+    const editorType = config.editor.type
+    let editor
+    if (editorType === 'DEFAULT') {
+      editor =
+        <MarkdownEditor
```
Comment (sota1235) [inline browser/main/Detail/MarkdownNoteDetail.js]: You can return on this line.

File: browser/main/Detail/MarkdownNoteDetail.js [hunk 3]
```diff
@@ -262,6 +263,32 @@ class MarkdownNoteDetail extends React.Component {
...
+        />
+    } else if (editorType === 'WYSIWYG') {
+      editor =
+        <WYSIWYGEditor
+          ref='content'
+          value={this.state.note.content}
+          config={config}
+          onChange={(e) => this.handleChange(e)} />
+    }
+    return editor
```
Comment (sota1235) [inline browser/main/Detail/MarkdownNoteDetail.js]: if `editorType` is invalid value, this function will return null
I think assign `MarkdownEditor` to `editor` as a default value is more safe

File: browser/main/lib/ConfigManager.js [hunk 4]
```diff
@@ -25,7 +25,8 @@ export const DEFAULT_CONFIG = {
     theme: 'default',
     showCopyNotification: true,
     disableDirectWrite: false,
-    defaultNote: 'ALWAYS_ASK' // 'ALWAYS_ASK', 'SNIPPET_NOTE', 'MARKDOWN_NOTE'
+    defaultNote: 'ALWAYS_ASK', // 'ALWAYS_ASK', 'SNIPPET_NOTE', 'MARKDOWN_NOTE'
```
Comment (sota1235) [inline browser/main/lib/ConfigManager.js]: [Comment]I feel we need flow 😢

File: browser/components/WYSIWYGEditor.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    this.setState({
+      value: props.value,
+      config: props.config
+    })
+  }
+
+  handleOnChange (e) {
+    this.setState({
+      value: e
+    }, () => { this.value = e })
```
Comment (sota1235) [inline browser/components/WYSIWYGEditor.js]: Do you need 2nd argument? It seems that `this.value` is not used.

File: browser/components/WYSIWYGEditor.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    fontSize: `${fontSize}px`
+  }
+}
+
+class WYSIWYGEditor extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      value: props.value,
+      config: props.config
```
Comment (sota1235) [inline browser/components/WYSIWYGEditor.js]: You don't need to set `config` state.
This component won't change `config`

File: browser/main/lib/ConfigManager.js (hunk 4 above)
Comment (sosukesuzuki) [reply browser/main/lib/ConfigManager.js]: I think so too 😢

File: browser/components/WYSIWYGEditor.js (hunk 5 above)
Comment (sosukesuzuki) [reply browser/components/WYSIWYGEditor.js]: Parent of this component(MarkdownNoteDetailComponent) is need `this.value`.
like below:
```
handleChange (e) {
  const { note } = this.state
  note.content = this.refs.content.value
  this.setState({note})
}
```

File: browser/components/WYSIWYGEditor.js (hunk 6 above)
Comment (sosukesuzuki) [reply browser/components/WYSIWYGEditor.js]: fix by c53542e 😃

File: browser/main/Detail/MarkdownNoteDetail.js (hunk 3 above)
Comment (sosukesuzuki) [reply browser/main/Detail/MarkdownNoteDetail.js]: 8a472b4 !

File: browser/main/Detail/MarkdownNoteDetail.js (hunk 2 above)
Comment (sosukesuzuki) [reply browser/main/Detail/MarkdownNoteDetail.js]: 91cebc2 👍

**Comment by sosukesuzuki** [comment]
It is so hard to implement WYSIWYG editor for us...
I'll close this pull request. :bow:

**Pull request closed without merging**